File: Frontend/manga-world/src/Model/PublisherManga.ts
import Manga from "./Manga";
import Publisher from "./Publisher";

class publisherManga {
    pId: number;
    mId: number;
    publisher: Publisher;
    manga: Manga;

    constructor(pId: number, mId: number, publisher: Publisher, manga: Manga) {
        this.pId = pId;
        this.mId = mId;
        this.publisher = publisher;
        this.manga = manga;
    }
}

export default publisherManga;

File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/PublisherManga.java
package com.amakakeru.mangaworld.Entity;

import com.amakakeru.mangaworld.Entity.IdClasses.PublisherMangaId;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@IdClass(PublisherMangaId.class)
@Table(name = "publisher_manga")
public class PublisherManga {

    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "p_id")
    private Long pid;

    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "m_id")
    private Long mid;

    @ManyToOne
    @JoinColumn(name = "p_id", insertable = false, updatable = false)
    private Publisher publisher;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;
}
